File: Lib_JS/src/schemaRoot.ts
import {EntitySchema} from "./entitySchema"
import {RelationSchema} from "./relationSchema"

export class SchemaRoot {

  public descriptionFormat: string = '';

  public timestampUtc: string = '';
  public semanticVersion: string = '';

  public entities: EntitySchema[] = [];

  public relations: RelationSchema[] = [];

}

File: dotnet/src/ModelDescription/SchemaRoot.cs
using System;
using System.Linq;

namespace System.Data.ModelDescription {

  /// <summary>
  /// (from 'FUSE-fx.ModelDescription')
  /// </summary>
  public class SchemaRoot {

    public string DescriptionFormat { get; set; } = "";

    public string TimestampUtc { get; set; } = "";

    public string SemanticVersion { get; set; } = "";

    public EntitySchema[] Entities { get; set; } = { };

    public RelationSchema[] Relations { get; set; } = { };

    public KnownValueRange[] KnownValueRanges { get; set; } = { };

    public string DesignerData { get; set; } = "";
  }

}
